dask_expr/_collection.py

```diff
@@ -1245,8 +1245,26 @@ def repartition(
                 Repartition(self, npartitions, divisions, force, partition_size, freq)
             )
 
-    def to_dask_dataframe(self, optimize: bool = True, **optimize_kwargs) -> _Frame:
-        """Convert to a dask-dataframe collection
+    def to_dask_dataframe(self, *args, **kwargs) -> _Frame:
+        """Convert to a legacy dask-dataframe collection
+
+        WARNING: This API is deprecated. Please use `to_legacy_collection`.
+
+        Parameters
+        ----------
+        optimize
+            Whether to optimize the underlying `Expr` object before conversion.
+        **optimize_kwargs
+            Key-word arguments to pass through to `optimize`.
+        """
+        warnings.warn(
+            "`to_dask_dataframe` is deprecated, please use `to_legacy_collection`.",
+            FutureWarning,
+        )
+        return self.to_legacy_collection(*args, **kwargs)
+
+    def to_legacy_collection(self, optimize: bool = True, **optimize_kwargs) -> _Frame:
```

Comment:
This is too internal imo, lets just use to_legacy_dataframe

Reply:
Yeah, I've always been a bit bothered by the (lack of?) Series/Dataframe distinction, but I agree that collection is probably worse. will use `to_legacy_dataframe`